### PR TITLE
3DR Control Zero H7 OEM Revision G support 

### PR DESCRIPTION
**Pull request by TOTON95**
Support added for 3DR Control Zero H7 OEM Revision G support

### CODE REVIEW

**Review by peterbarker — CHANGES_REQUESTED** [review]
commit list needs a bit of cleanup; should only be two commits, one for the bootloader binaries and one for the hwdef.

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (TOTON95) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md]: Suggestion Applied

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (TOTON95) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md]: Suggestion Applied

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (TOTON95) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md]: Suggestion Applied

**Comment by lgarciaos** [comment]
Hi @Hwurzburg, please let us know what you think of the new README sections.

**Comment by lgarciaos** [comment]
@peterbarker, I think we have made all the requested changes. Please let us know if we can improve it any way.

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:67]: currently this is gps2 by default, you would need the hwdef define its protocol to SerialProtocol_FrSky_SPort_Passthrough to get frsky telemetry

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:95]: you have not defined any outputs bdshot capable

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:67]: you dont show this uart as being pinned out in the connection diagram

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:77]: I cant see which uart thi is being input to...none have been set to SerialProtocol_RCIN by the hwdef

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:81]: this not an rc protocol...if yu mean FPort, the it requires a half duplex serial option on the uart port and connection to the tx oin or rx/tx swap option also

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

**Comment by TOTON95** [comment]
@Hwurzburg Thanks for the review, I'll address your comments ASAP 🙌

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (lgarciaos) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:95]: @peterbarker, do you think it makes sense to have 2 separate targets to support BIDIR?  What are the caveats? If there aren't any let's have it as default.

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (TOTON95) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:67]: This is going to be replaced

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (TOTON95) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:95]: @Hwurzburg Can you help us to determine if we have to make 2 targets for BIDIR and w/o BIDIR?

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (TOTON95) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:67]: Changed to keep default mapping

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:95]: that would be @andypiper...hes the bidir guru....but since its an H7, there should be a way to have at least 4 bidir outputs on a single target

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:67]: default mapping makes Serial 4 gps2, not frsky...you can set it to frsky (passthrough) in the hwdef (preferably) or the defaults.parm if that what you want

Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:67]: it still needs  to be documented as to where this uart is pinned out

File: libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/3DRControlZeroG/README.md:77]: ok, found the INT pin

**Review by Hwurzburg — APPROVED** [review]
still think its a waste not to have at least 4 outputs bidir  since it is an H7

**Comment by TOTON95** [comment]
@Hwurzburg About the BIDIR outputs, @lgarciaos just sent a board to @andyp1per. So we might have some news from Andy about it around next week.

**Comment by Hwurzburg** [comment]
@TOTON95 please detail the last changes you made in that forced push....it could invalidate my approval....without you detailing the changes I would have to go thru a complete re-review from scratch which could be a large waste of time.....I would rather just review the changes you made for that last push

**Comment by TOTON95** [comment]
@Hwurzburg Sure my bad. I had to modify the hwdef.dat file to remove the following lines: 
  
```c++
 
define HAL_GPIO_LED_ON 0
define HAL_GPIO_LED_OFF 1 
 
```   
 
To comply with Peter's PR #26642 and get the binary built successfully.  I already tested in local as well.

**Comment by TOTON95** [comment]
Also, did a rebase from current ArduPilot's master branch

**Comment by Hwurzburg** [comment]
tnx

**Review by andyp1per — APPROVED** [review]

**Comment by tridge** [comment]
@peterbarker will merge after fixing commit msgs

**Comment by TOTON95** [comment]
@peterbarker I just force-pushed the changes that you suggested, so the message and commits match your criteria, thanks for your temp branch and double-checking this last operation. I will let you know the next time I open a new PR so we can make sure that you guys have a way to modify and force push as needed.

**Comment by TOTON95** [comment]
Thank you!

**Comment by rmackay9** [comment]
This is included in 4.5.6-beta1

**Pull request merged**
